File: Demo-NodeJs/app/domain/Person.ts
import {Task} from "./Task";

export class Person {
    id : number;
    firstName : string;
    lastName : string;
    salary : number;
    dob : string;
    status : string;
    tasks : Task[];

    constructor(firstName: string, lastName: string, salary: number, dob: string, status: string, tasks: Task[]) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.salary = salary;
        this.dob = dob;
        this.status = status;
        this.tasks = tasks;
    }
}

File: learn_criteria/src/main/java/com/fintech/domain/Person.java
package com.fintech.domain;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@EqualsAndHashCode
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String firstName;
    private String lastName;
    @JsonFormat(pattern = "yyyy-MM-dd")
    private Date dob;
    @Enumerated(value = EnumType.STRING)
    private Status status;

    public enum  Status {
        ACTIVE, INACTIVE
    }
}
